### PR TITLE
Limit registrations to a specific email domain

### PR DESCRIPTION
**Pull request by johannesst**
Hello, I noticed your comment // FEATURE: allow only from specific email domain and tried to hack such a feature by my own. Its quite hacky and needs further testing (although my tests went fine) and more translations. You can configure the Domains in the settings panel by entering a semicolon-seperated list (e.g "google.com;tu-bs.de;tu-braunschweig.de). Then only people with a Mailadress with these domains can register. If you delete the list (aka changing it to "") everybody can register again. Please inform me, If you encounter any problems within my code. It's propobly buggy since it's the work of a night of coding. Regards

### CODE REVIEW

File: controller/registercontroller.php [hunk 1]
```diff
@@ -67,7 +67,7 @@ public function askEmail($errormsg, $entered) {
 	public function validateEmail() {
 		$email = $this->request->getParam('email');
 		if ( !filter_var($email, FILTER_VALIDATE_EMAIL) ) {
-			return new TemplateResponse('', 'error', array(array('error' => $this->l10n->t('Email address you entered is not valid'))), 'error');
+		//	return new TemplateResponse('', 'error', array(array('error' => $this->l10n->t('Email address you entered is not valid'))), 'error');
```
Comment (pellaeon) [inline controller/registercontroller.php]: Oops, my bad, you could just delete the line.

File: js/settings.js [hunk 2]
```diff
@@ -1,9 +1,10 @@
 $(document).ready(function() {
 	function saveSettings() {
-		var post = $('#registered_user_group').serialize();
-
+		var post = $('#registration').serialize();
+		console.log(post);
```
Comment (pellaeon) [inline js/settings.js]: Please remove the debug line

Comment (johannesst) [inline js/settings.js]: Done

File: controller/registercontroller.php (hunk 1 above)
Comment (johannesst) [inline controller/registercontroller.php]: It's now gone for good

**Comment by pellaeon** [comment]
Could you add explanation regarding the semicolon-separated configuration format in the admin section?

**Comment by pellaeon** [comment]
When I save the configuration values, I get redirected to `index.php/settings/admin?registered_user_group=test1&allowed_domains=cnmc.tw#goto-registration`.
Do you know why?

File: controller/registercontroller.php [hunk 3]
```diff
@@ -94,8 +93,34 @@ public function validateEmail() {
...
+				}
+					
+			}
+			// $allowed still false->return error message
+			if ( $allowed === false ) {
+				return new TemplateResponse('registration', 'domains', ['domains' =>
+					$domains
+				], 'guest');
+			}
+		}//else var_dump($allowed_domains);
```
Comment (pellaeon) [inline controller/registercontroller.php]: Please remove debug lines.

File: controller/settingscontroller.php [hunk 4]
```diff
@@ -21,6 +21,7 @@
 use \OCP\IL10N;
 use \OCP\IConfig;
 use \OCP\IUser;
+use \OCP\AppFramework\Http\JSONResponse;
```
Comment (pellaeon) [inline controller/settingscontroller.php]: `JSONResponse` is not used, could you remove it?

File: controller/settingscontroller.php [hunk 5]
```diff
@@ -37,13 +38,27 @@ public function __construct($appName, IRequest $request, IL10N $l10n, IConfig $c
...
+	public function admin($registered_user_group, $allowed_domains) {
+		if ( ( $allowed_domains==='' ) || ( $allowed_domains === NULL ) ){
+			$this->config->deleteAppValue($this->appName, 'allowed_domains');
+		}else{
+			/*
+			$new_domains='';
+			$domains=explode($allowed_domains,';');
+			foreach ( $domains as $domain) {
+				if( (strlen($domain)i >= 3) && (strpos($domain,'.'
+			}*/
```
Comment (pellaeon) [inline controller/settingscontroller.php]: Are these left with a reason?

**Comment by johannesst** [comment]
You wrote:
> When I save the configuration values, I get redirected to index.php/settings/admin?registered_user_group=test1&allowed_domains=cnmc.tw#goto-registration.
> Do you know why?

Nope, I couldn't reproduce it. This should never happen, since this is an url generated by a GET-Request. The app uses afik only POST-Requests, so this redirect should never appear. I used Chromium and firefox for my tests. Which browser did you use for testing? I can try to reproduce the error.

**Comment by pellaeon** [comment]
What version of ownCloud are you on?

**Comment by pellaeon** [comment]
Ah, I see it now, if I press Enter after entering the values the browser will send a GET request.
I'll merge your PR and solve that later.

**Pull request merged**